src/shims/posix/sync.rs

```diff
@@ -652,7 +660,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn pthread_condattr_destroy(&mut self, attr_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
-        condattr_set_clock_id(this, attr_op, ScalarMaybeUninit::Uninit)?;
+        condattr_get_clock_id(this, attr_op)?.check_init()?;
```

Comment:
This is just as an extra UB check, right? Looks good, but please add a comment explaining it.

Reply:
Also, can you check what we do for the other `destroy` functions in this file? They should all be consistent.

Reply:
Doesn't look like anywhere else checks for uninit before they set it to be uninit.

I'll adjust `pthread_mutexattr_destroy`, `pthread_mutex_destroy`, `pthread_rwlock_destroy`, `pthread_cond_destroy` to all check for uninit using this pattern then.

My man pages say destroying something that's already been destroyed is UB for *all* of those, and I'm assuming that's universal. Will add `compile-fail` tests for those too.